jlh/WeatherApp/app/src/main/java/com/e_conomic/weatherapp/ui/ForecastListAdapter.kt

```diff
@@ -30,9 +31,17 @@ class ForecastListAdapter(val weekForecast: ForecastList, val itemClick: (Foreca
 
         fun bindForecast(forecast: Forecast) {
             with(forecast) {
-                Picasso.with(itemView.ctx).load(iconUrl).into(itemView.icon)
+
+                if (iconUrl == ForecastDataMapper.NO_ICON) {
+                    itemView.icon.setImageResource(R.mipmap.ic_launcher)
+                } else {
+                    Picasso.with(itemView.context).load(iconUrl).into(itemView.icon)
+                }
+
                 itemView.date.text = date
-                itemView.description.text = description
+                itemView.description.text = if (description == ForecastDataMapper.NO_DESCRIPTION)
+                    App.instance.resources.getString(R.string.default_weather_description)
```

Comment:
Now I use the application singleton 🌝 Not that nice though.. can you do it differently?